File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        double r1 = (double) common[1] / common[0];
        double r2 = (double) common[2] / common[1];
        if(r1 % 1 == 0 && r2 % 1 == 0 && (int)r1 == (int)r2) {
            return (int)r1 * common[common.length - 1];
        }
        int d1 = common[1] - common[0];
        int d2 = common[2] - common[1];
        if(d1 == d2) return common[common.length - 1] + d1;
        return 0;
    }
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.py
def solution(common): 
    if (common[1] - common[0]) == (common[2] - common[1]):
        return common[len(common) -1] + (common[1] - common[0])
    elif int(common[1]/common[0]) == int(common[2]/common[1]):
        return common[len(common) -1] * int(common[1]/common[0])
    
        
        